src/content/reference/cluster-upgrades/index.md

```diff
@@ -0,0 +1,167 @@
+
+## How upgrades work
+
+All levels of upgrades, patch, minor, or major, are happening at runtime.
+Your Kubernetes workloads should continue to work as expected (given they meet a few requirements, as explained further below).
+
+In particular this means:
+
+- The master node will be taken down, resulting in the Kubernetes API being shortly unavailable.
+- The worker nodes will be drained and taken down, resulting in pods being rescheduled to other nodes.
+
+**Note**: We have high-availability masters (multiple master nodes) on our roadmap, which will help prevent Kubernetes API downtimes during upgrades.
+
+---
+
+## Remainders of outline
+
+* Testing (e2e and conformance)
+
+## Releases
+
+* old clusters will be force upgraded)
+* Explain versions of subcomponents
+
+## Upgrades
+
+* Explain upgrades in each provider. 
+* Maybe include a highlevel diagram of how cluster-operator, node-operator and provider specific operators deal with upgrades
+
+### AWS
+
+* Cloudformation upgrade. Master first. Then rolling update of the ASG. 
+    * more control over the ASG update is still in progress. goal here: new nodes are created first, until they are in the kubernetes api and ready and then the old nodes are drained and terminated.
+
+### Azure
+
+* Azure resource manager. All nodes get reimaged and the updated including the master. Master first.
+    * This happens one by one. Nodes are drained and then rebooted with the new image and configuration
+    * A delete has been discussed here instead of update of existing nodes. But this is not clear yet.
+
+### KVM
+
+* kvm-operator updating one node after each other.
+    * Master first?
+    * Nodes are drained and then the pod is replaced with a new version.
+
+## How to prepare your workloads
+
+### Scale up workloads
+
+Make sure you have 2 or more replicas for al your deployments. For critical components you might want to go with more than just the bare minimum of 2.
+
+You can adjust this depending on your environments, e.g. running 2-3 replicas in `DEV`, and 5 or more in `PROD`.
+
+In case you are using Horizontal Pod Autoscaler, above recommendations should depict your minimum replica setting.
+
+### Manage disruption budgets
+
+Configure [PodDisruptionBudgets](https://kubernetes.io/docs/tasks/run-application/configure-pdb/) for all your deployments. This tells Kubernetes to keep a minimum amount of Pods running at all times and is respected by the draining/eviction mechanisms during upgrades.
+
+### Make rescheduling graceful
+
+* Have well implemented live and ready probes
+* Set proper values for initialDelaySeconds and termincationGracePeriod (related to previous point)
+
+### Set scheduling priorities
+
+* Consider using pod priority preemption to ensure critical pods run always
+
+### Avoid ephemeral resources
+
+* Avoid using pods without backed up resource (deployment, daemonset, ...)
+* Avoid using local storage (or use it as cache)
+
```

Comment:
I would explain explicitly that if you use local storage you will lose the information. Same for the bare pods

Reply:
sorry, your comment was on the outdated doc, see above